odd-platform-api/src/test/java/org/opendatadiscovery/oddplatform/repository/ReactiveDataQualityRunsRepositoryTest.java

```diff
@@ -0,0 +1,214 @@
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.opendatadiscovery.oddplatform.model.Tables.DATA_ENTITY_TASK_LAST_RUN;
+
+@TestInstance(TestInstance.Lifecycle.PER_CLASS)
+@TestExecutionListeners(listeners = {DependencyInjectionTestExecutionListener.class})
+public class ReactiveDataQualityRunsRepositoryTest extends BaseIntegrationTest {
+    public static final String DATA_QUALITY_TEST_ATTRIBUTE = """
+            {
+              "DATA_QUALITY_TEST": {
+                "expectation": {
+                  "category": "{type}"
+                }
+              }
+            }""";
+    public static final int NUMBER_OF_ASSERTION = 5;
+    public static final int NUMBER_OF_SCHEMA_CHANGES = 3;
+    public static final int NUMBER_OF_COLUMN_ANOMALY = 2;
+    public static final int NUMBER_OF_FRESHNESS_ANOMALY = 4;
+    public static final int NUMBER_OF_VOLUME_ANOMALY = 1;
+
+    public static final Map<DataQualityCategory, Pair<List<DataEntityRunStatus>, Integer>> EXPECTED_VALUES = Map.of(
+            DataQualityCategory.ASSERTION, Pair.of(List.of(DataEntityRunStatus.SUCCESS), NUMBER_OF_ASSERTION),
+            DataQualityCategory.SCHEMA_CHANGE,
+            Pair.of(List.of(DataEntityRunStatus.SUCCESS, DataEntityRunStatus.ABORTED), NUMBER_OF_SCHEMA_CHANGES),
+            DataQualityCategory.COLUMN_VALUES_ANOMALY,
+            Pair.of(List.of(DataEntityRunStatus.SUCCESS,
+                    DataEntityRunStatus.FAILED), NUMBER_OF_COLUMN_ANOMALY),
+            DataQualityCategory.FRESHNESS_ANOMALY,
+            Pair.of(List.of(DataEntityRunStatus.SKIPPED,
+                    DataEntityRunStatus.ABORTED), NUMBER_OF_FRESHNESS_ANOMALY),
+            DataQualityCategory.VOLUME_ANOMALY,
+            Pair.of(List.of(DataEntityRunStatus.SKIPPED, DataEntityRunStatus.FAILED), NUMBER_OF_VOLUME_ANOMALY)
+    );
+    @Autowired
+    private ReactiveDataQualityRunsRepository repository;
+    @Autowired
+    private ReactiveDataEntityTaskRunRepository dataEntityTaskRunRepository;
+    @Autowired
+    private ReactiveDataEntityRepository dataEntityRepository;
+    @Autowired
+    private ReactiveDataQualityTestRelationRepository dataQualityTestRelationRepository;
+    @Autowired
+    private JooqReactiveOperations jooqReactiveOperations;
+    @Autowired
+    private DataQualityCategoryMapper testsMapper;
+    @Autowired
+    private TablesDashboardMapper tablesDashboardMapper;
+
+    @BeforeAll
+    public void generateEntitiesAndTest() {
+        generateTestsData();
+    }
+
+    @Test
+    @DisplayName("Test get Latest Data Quality Runs")
+    public void testGetLatestDataQualityRunsResults() {
+        final List<DataQualityCategoryResults> testResults =
+                testsMapper.mapToDto(repository.getLatestDataQualityRunsResults()
+                        .collectList()
+                        .block());
+
+        dataQualityTestAssertion(testResults);
+    }
```

Comment:
I'm not sure if it's better or not but I just take a look through other repositories tests and my concern is if we are dealing with streams and repositories are streams, so would it be more consistent and logical to treat and test them in the same way. I've just took example from other repositories tests
```suggestion
    @Test
    @DisplayName("Test get Latest Data Quality Runs")
    public void testGetLatestDataQualityRunsResults() {
       repository.getLatestDataQualityRunsResults()
            .as(StepVerifier::create)
            # some testing steps. Not sure how to make it better
            # pseudo step.
            .assertNext(testResults -> dataQualityTestAssertion(testResults))
            .verifyComplete();
    }
```

Reply:
done